Examples/text-size-test.lisp

```diff
@@ -13,182 +14,233 @@
 ;;; Library General Public License for more details.
 ;;;
 ;;; You should have received a copy of the GNU Library General Public
-;;; License along with this library; if not, write to the 
-;;; Free Software Foundation, Inc., 59 Temple Place - Suite 330, 
+;;; License along with this library; if not, write to the
+;;; Free Software Foundation, Inc., 59 Temple Place - Suite 330,
 ;;; Boston, MA  02111-1307  USA.
 
-(in-package :clim-demo)
+(cl:in-package #:clim-demo)
+
+;;; State
+
+(defclass state ()
+  ((%text        :initarg  :text
+                 :accessor text)
+   (%text-family :initarg  :text-family
+                 :accessor text-family)
+   (%text-face   :initarg  :text-face
+                 :accessor text-face)
+   (%text-size   :initarg  :text-size
+                 :accessor text-size*)
+   (%rectangle   :initarg  :rectangle
+                 :type     (member :text-size :text-bounding-rectangle)
+                 :accessor rectangle)
+   (%hook        :initarg  :hook
+                 :accessor hook
+                 :initform nil)))
+
+(defmethod maybe-run-hook ((state state))
+  (alexandria:when-let ((hook (hook state)))
+    (funcall hook state)))
+
+(defmethod (setf text) :after ((new-value t) (state state))
+  (maybe-run-hook state))
+
+(defmethod (setf text-family) :after ((new-value t) (state state))
+  (maybe-run-hook state))
+
+(defmethod (setf text-face) :after ((new-value t) (state state))
+  (maybe-run-hook state))
+
+(defmethod (setf text-size*) :after ((new-value t) (state state))
+  (maybe-run-hook state))
+
+(defmethod (setf rectangle) :after ((new-value t) (state state))
+  (maybe-run-hook state))
+
+(defmethod text-style ((state state))
+  (make-text-style (text-family state) (text-face state) (text-size* state)))
+
+;;; Canvas
+
+(defclass canvas (application-pane)
+  ((%state :initarg :state
+           :reader  state)))
+
+(defmethod initialize-instance :after ((instance canvas) &key state)
+  (setf (hook state) (lambda (state)
+                       (declare (ignore state))
+                       (repaint-sheet instance (sheet-region instance)))))
+
+(defmethod resize-sheet :after ((sheet canvas) (width t) (height t))
+  (repaint-sheet sheet (sheet-region sheet)))
```

Comment:
is there a reason why you specialize on T? if not, please remove it (estabilished code convention is to leave unspecialized arguments be).

Reply:
Changed.